freud/common.py

```diff
@@ -56,3 +56,246 @@ def convert_box(box):
+        pmft_arr[np.isinf(pmft_arr)] = np.nan
+        dx = (2.0 * 3.0) / pmft.n_bins_X
+        dy = (2.0 * 3.0) / pmft.n_bins_Y
+        nan_arr = np.where(np.isnan(pmft_arr))
+        for i in range(pmft.n_bins_X):
+            x = -3.0 + dx * i
+            for j in range(pmft.n_bins_Y):
+                y = -3.0 + dy * j
+                if ((x*x + y*y < 1.5) and (np.isnan(pmft_arr[j, i]))):
+                    pmft_arr[j, i] = 10.0
+        w = int(2.0 * pmft.n_bins_X / (2.0 * 3.0))
+        center = int(pmft.n_bins_X / 2)
+
+        # Get the center of the histogram bins
+        pmft_smooth = gaussian_filter(pmft_arr, 1)
+        pmft_image = np.copy(pmft_smooth)
+        pmft_image[nan_arr] = np.nan
+        pmft_smooth = pmft_smooth[center-w:center+w, center-w:center+w]
+        pmft_image = pmft_image[center-w:center+w, center-w:center+w]
+        x = pmft.X
+        y = pmft.Y
+        reduced_x = x[center-w:center+w]
+        reduced_y = y[center-w:center+w]
+
+        # Plot figures
+        fig = Figure(figsize=(12, 5), facecolor='white')
+        values = [-2, -1, 0, 2]
+        norm = Normalize(vmin=-2.5, vmax=3.0)
+        n_values = [norm(i) for i in values]
+        colors = viridis(n_values)
+        colors = colors[:, :3]
+        # verts = make_polygon(sides=6, radius=0.6204)
+        lims = (-2, 2)
+        ax0 = fig.add_subplot(1, 2, 1)
+        ax1 = fig.add_subplot(1, 2, 2)
+        for ax in (ax0, ax1):
+            # commented out since in general, particles are not hexagonal
+            # ax.contour(reduced_x, reduced_y, pmft_smooth,
+            #            [9, 10], colors='black')
+            # ax.contourf(reduced_x, reduced_y, pmft_smooth,
+            #             [9, 10], hatches='X', colors='none')
+            # ax.plot(verts[:,0], verts[:,1], color='black', marker=',')
+            # ax.fill(verts[:,0], verts[:,1], color='black')
+            ax.set_aspect('equal')
+            ax.set_xlim(lims)
+            ax.set_ylim(lims)
+            ax.xaxis.set_ticks([i for i in range(lims[0], lims[1]+1)])
+            ax.yaxis.set_ticks([i for i in range(lims[0], lims[1]+1)])
+            ax.set_xlabel(r'$x$')
+            ax.set_ylabel(r'$y$')
+
+        ax0.set_title('PMFT Heat Map')
+        im = ax0.imshow(np.flipud(pmft_image),
+                        extent=[lims[0], lims[1], lims[0], lims[1]],
+                        interpolation='nearest', cmap='viridis',
+                        vmin=-2.5, vmax=3.0)
+        ax1.set_title('PMFT Contour Plot')
+        ax1.contour(reduced_x, reduced_y, pmft_smooth,
+                    [-2, -1, 0, 2], colors=colors)
+
+        fig.subplots_adjust(right=0.85)
+        cbar_ax = fig.add_axes([0.88, 0.1, 0.02, 0.8])
+        fig.colorbar(im, cax=cbar_ax)
+        f = io.BytesIO()
+        canvas = FigureCanvasAgg(fig) # noqa F841
+        fig.savefig(f, format='png')
+        return f.getvalue()
+
+
+def draw_voronoi(box, cells, color_by_sides=False):
```

Comment:
@klarh I was thinking about the related Voronoi-drawing code in `plato` - anything we should consider when creating this IPython representation of `freud` Voronoi objects?